### PR TITLE
[SPARK-42043][CONNECT][TESTS] Fix connect jar finding issue for scala 2.13

### PR DESCRIPTION
**Pull request by zhenlineo**
### What changes were proposed in this pull request?

Fix the server jar finding error for Scala 2.13 distribution for Scala client E2E tests. See https://github.com/apache/spark/pull/39541

### Why are the changes needed?

Bug fix

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by zhenlineo** [comment]
cc @dongjoon-hyun Let me know if there is more env that need to be addressed.

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/util/RemoteSparkSession.scala [hunk 1]
```diff
@@ -188,7 +189,7 @@ trait RemoteSparkSession
 
   override def afterAll(): Unit = {
     try {
-      spark.close()
+      if (spark != null) spark.close()
```
Comment (dongjoon-hyun) [inline connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/util/RemoteSparkSession.scala:192]: Is this related?

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/util/RemoteSparkSession.scala [hunk 2]
```diff
@@ -118,12 +118,13 @@ object SparkConnectServerUtils {
...
+      s"Fail to locate the spark connect server target folder: '${parentDir.getCanonicalPath}'. " +
         s"SPARK_HOME='${new File(sparkHome).getCanonicalPath}'. " +
-        "Make sure system property `SPARK_HOME` is set correctly.")
+        "Make sure the spark connect server jar has been built " +
+        "and the system property `SPARK_HOME` is set correctly.")
     val jars = recursiveListFiles(parentDir).filter { f =>
       // SBT jar
-      (f.getParent.endsWith("scala-2.12") &&
+      (f.getParentFile.getName.startsWith("scala-") &&
         f.getName.startsWith("spark-connect-assembly") && f.getName.endsWith("SNAPSHOT.jar")) ||
```
Comment (dongjoon-hyun) [inline connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/util/RemoteSparkSession.scala:128]: `SNAPSHOT.jar` pattern should not be used, @zhenlineo .

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you for making a fix.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master. Thank you, @zhenlineo .

**Comment by LuciferYang** [comment]
Thanks @zhenlineo for fix this
